### PR TITLE
[DD-1111] applied GDM JSON stream processing + some performance improvements ...

### PR DESCRIPTION
**Pull request by boferri**
... re. schema by example generation

### CODE REVIEW

File: controller/src/test/java/org/dswarm/controller/resources/resource/test/DataModelsResourceTest2.java [hunk 1]
```diff
@@ -214,6 +219,56 @@ public void testCSVDataUpdate2() throws Exception {
 		Assert.assertTrue(data.isPresent());
 		Assert.assertNotNull(assoziativeJsonArray);
 
+		final String expectedResult = DMPPersistenceUtil.getResourceAsString("dd-762.expected.delta-result.json");
+		final String actualResult = objectMapper.writeValueAsString(assoziativeJsonArray);
+
+		// TODO: do proper comparison
+		Assert.assertEquals(expectedResult.length(), actualResult.length());
```
Comment (knutwalker) [inline controller/src/test/java/org/dswarm/controller/resources/resource/test/DataModelsResourceTest2.java:226]: JsonAssert?

File: persistence/src/main/java/org/dswarm/persistence/service/internal/graph/InternalGDMGraphService.java [hunk 2]
```diff
@@ -552,6 +553,120 @@ public void deleteObject(final String dataModelUuid) throws DMPPersistenceExcept
...
+//				}
+
+				gdmModel.setRecordURIs(realModel.getResourceURIs());
+			}
+		}
+
+		LOG.debug("determined record resources for data model '{}'", finalDataModel.getUuid());
+	}
+
+	private Optional<Boolean> determineMissingRecordsFlag(final UpdateFormat updateFormat) throws DMPPersistenceException {
```
Comment (knutwalker) [inline persistence/src/main/java/org/dswarm/persistence/service/internal/graph/InternalGDMGraphService.java:654]: Optional<Boolean> seems like overkill, couldn't this just be boolean? You never return Optional[None], anyway.

**Comment by knutwalker** [comment]
:+1: in general, but let's figure out this jenkins issue :wink:

File: persistence/src/main/java/org/dswarm/persistence/service/internal/graph/InternalGDMGraphService.java (hunk 2 above)
Comment (boferri) [inline persistence/src/main/java/org/dswarm/persistence/service/internal/graph/InternalGDMGraphService.java:654]: yep, we should simplify later.

File: controller/src/test/java/org/dswarm/controller/resources/resource/test/DataModelsResourceTest2.java (hunk 1 above)
Comment (boferri) [inline controller/src/test/java/org/dswarm/controller/resources/resource/test/DataModelsResourceTest2.java:226]: no, or when we like to apply it's a bit more complicated here. the records uris will always be generated and differ then from the execepted result. so one always need to replace them before comparing the JSON with JsonAssert.

**Pull request merged**